mgmtfn/dockplugin/netDriver.go

```diff
@@ -742,10 +742,28 @@ func createNetworkHelper(networkID string, tag string, IPv4Data, IPv6Data []driv
 	return nil
 }
 
-// deleteNetworkHelper removes the association between docker network and contiv network
+// deleteNetworkHelper removes the association between docker network
+// and contiv network. We have to remove docker network state before
+// remove network in contiv.
 func deleteNetworkHelper(networkID string) error {
+	dnet, err := docknet.FindDocknetByUUID(networkID)
+	if err == nil {
+		// delete the dnet oper state
+		err = docknet.DeleteDockNetState(dnet.TenantName, dnet.NetworkName, dnet.ServiceName)
+		if err != nil {
+			msg := fmt.Sprintf("Could not delete docknet for nwID %s: %s", networkID, err.Error())
+			log.Errorf(msg)
+			return errors.New(msg)
+		}
+		log.Infof("Deleted docker network mapping for %v", networkID)
+	} else {
+		msg := fmt.Sprintf("Could not find Docker network %s: %s", networkID, err.Error())
```

Comment:
fmt package takes care of calling Error()
fmt.Sprintf("xxx %s", err)

Reply:
I am going to stick with the current implementation

Reply:
i understand now that due to other bad behavior (not erroring in v2plugin mode when requesting to create the network) that we can get out of sync, I'm just looking for a ticket to block the netctl when in v2plugin mode, then we can clean up this lax behavior here